engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -189,9 +189,14 @@ func (jm *JobMaster) OnWorkerOnline(worker framework.WorkerHandle) error {
 
 // handleOnlineStatus is used by OnWorkerOnline and OnWorkerStatusUpdated.
 func (jm *JobMaster) handleOnlineStatus(worker framework.WorkerHandle) error {
+	extBytes := worker.Status().ExtBytes
+	if len(extBytes) == 0 {
```

Comment:
maybe we can remove the usage of `OnWorkerOnline` callback, only use `OnWorkerStatusUpdated`? That is to say, because DM needs bussiness online message with worker bound information, plain framework online is no use

Reply:
What if `OnWorkerStatusUpdated ` is not called because the message is discarded for some reasons, like `buffer full`?

Reply:
worker will be regard as offline after 2*HeartbeatInterval. https://github.com/pingcap/tiflow/blob/822a79f6e2e2619d00df4d6b70572cf5064b4ba3/engine/jobmaster/dm/worker_manager.go#L159

Reply:
https://github.com/pingcap/tiflow/blob/822a79f6e2e2619d00df4d6b70572cf5064b4ba3/engine/jobmaster/dm/worker_manager.go#L159